app/javascript/components/custom-url-tabs/index.jsx

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { Tab, Tabs, Modal } from 'carbon-components-react';
+import { checkForFormChanges } from './helper';
+
+const CustomURLTabs = ({
+  tabs, path, currentTab, checkForChanges,
+}) => {
+  const [tabsList, setTabsList] = useState([]);
+  const [selectedTab, setSelectedTab] = useState(0);
+  const [showConfirm, setShowConfirm] = useState(false);
+  const [url, setUrl] = useState();
+
+  const onTabClick = (id) => {
+    if (currentTab !== id) {
+      if (checkForChanges && checkForFormChanges()) {
+        setUrl(`${path}${id}?uib-tab=${id}`);
+        setShowConfirm(true);
+        document.getElementById(id).parentElement.classList.add('bx--tabs--scrollable__nav-item--selected');
+      } else {
+        window.location.replace(`${path}${id}?uib-tab=${id}`);
+      }
+    }
+  };
+
+  const fixTabStyling = () => {
+    const selectedTabs = [];
+    document.getElementsByClassName('bx--tabs--scrollable__nav-item--selected').forEach((element) => {
+      selectedTabs.push(element);
+    });
+    selectedTabs.forEach((tab) => {
+      tab.classList.remove('bx--tabs--scrollable__nav-item--selected');
+    });
+    document.getElementsByClassName('selected')[0].classList.add('bx--tabs--scrollable__nav-item--selected');
+  };
+
+  useEffect(() => {
+    const tempTabs = [];
+    tabs.forEach((tab, index) => {
+      if (tab[0] === currentTab) {
+        setSelectedTab(parseInt(index, 10));
+      }
+
+      tempTabs.push(
+        <Tab
+          key={tab[0]}
+          id={tab[0]}
+          className={`${tab[0]}` === currentTab ? 'selected' : 'not-selected'}
+          label={__(tab[1])}
+          onClick={() => onTabClick(tab[0])}
+        />
+      );
+    });
+    setTabsList(tempTabs);
+  }, []);
+
+  useEffect(() => {
+    if (showConfirm) {
+      document.getElementsByClassName('selected')[0].classList.remove('bx--tabs--scrollable__nav-item--selected');
+    } else if (document.getElementsByClassName('selected').length > 0) {
+      fixTabStyling();
+    }
+  }, [showConfirm]);
+
+  return (
+    <div>
```

Comment:
We do need it since if we remove it there will be 2 parent elements which are the modal and tablist. This will cause an error so we need to wrap them in a parent div